### PR TITLE
Add trakt ratings

### PR DESCRIPTION
**Pull request by Razzeee**
This is wip due to needed changes in the dependencies. But works fine with those.

I'm not really happy about us calling trakt even if ratings are not used, same goes for imdb ratings (unless you already changed that)

Will likely revisit that part.

Here is what we're waiting for https://github.com/fuzeman/trakt.py/pull/68

### CODE REVIEW

**Comment by rmrector** [comment]
> I'm not really happy about us calling trakt even if ratings are not used, same goes for imdb ratings (unless you already changed that)

IMDB ratings are always used / persisted if they are added, "default" just sets the one returned when a specific source isn't named - for skins `ListItem.Rating` pulls up default vs `ListItem.Rating(imdb)` shows the IMDB rating even when TMDB is the default.

I can agree there should be on/off switches for both, though.

**Comment by Razzeee** [comment]
I'm well aware of the data structure, was not thinking about it being nice to set both, but still, it seems pretty wasteful. And the trakt guys might not like us sending traffic there way for "no reason".

**Comment by Razzeee** [comment]
Our dependency has been updated, I will get this into shape as soon as possible. 
https://github.com/fuzeman/trakt.py/releases/tag/3.1.0

**Comment by Razzeee** [comment]
Dependency has been merged, but code seems to have been shuffled completely. Should I check if I can align it or can you do it?

**Comment by rmrector** [comment]
I'd prefer you give it a go - work has been heavy and it is not getting better very fast.

To set it up, move `tracktratings.py` to [python/lib/tmdbscraper](https://github.com/xbmc/metadata.themoviedb.org.python/tree/master/python/lib/tmdbscraper), but return a dict shaped like [`imdbratings.py`](https://github.com/xbmc/metadata.themoviedb.org.python/blob/master/python/lib/tmdbscraper/imdbratings.py#L32) - with no top250, "ratings" will be the only top-level key. Combine them with the original details in [scraper.py](https://github.com/xbmc/metadata.themoviedb.org.python/blob/master/python/scraper.py#L79) with `combine_scraped_details_info_and_ratings`, then update the [configuration](https://github.com/xbmc/metadata.themoviedb.org.python/blob/master/python/scraper_datahelper.py#L43) to add "trakt" to the possible `default_rating` values.

**Comment by Razzeee** [comment]
Updated and seems to work fine on the limited test set I have on my laptop.

File: python/scraper.py [hunk 1]
```diff
@@ -86,7 +95,10 @@ def get_details(input_uniqueids, handle):
     add_artworks(listitem, details['available_art'])
 
     for rating_type, value in details['ratings'].items():
-        listitem.setRating(rating_type, value['rating'], value['votes'], value['default'])
+        if 'votes' in value:
+            listitem.setRating(rating_type, value['rating'], value['votes'], value['default'])
+        else:
+            listitem.setRating(rating_type, value['rating'], value['default'])
```
[the remark this answers is not in the document]
Comment (Razzeee) [reply python/scraper.py]: I guess there is a typo in your recommendation?

**Comment by Razzeee** [comment]
Made the suggested changes. Feel free to squash them.

File: python/scraper.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (rmrector) [reply python/scraper.py]: [Not a typo](https://codedocs.xyz/xbmc/xbmc/group__python__xbmcgui__listitem.html#gad1fa7da3c659e70f53abc32ff3526f8e). Sorry I didn't mention that - I took a look at the C++ code to make sure the _docs_ didn't have a typo.

[the remark this answers is not in the document]
Comment (Razzeee) [reply python/scraper.py]: Aligned the code. Should we look at changing that typo in kodi?

**Comment by rmrector** [comment]
Thanks. Looks good and a quick runtime test works well here. A few unit tests are broken, but that's all still early unfinished manual stuff so I'll catch those up.

I'll have some time at the end of the month, so I'll work on minting a new version of the scraper.

**Pull request merged**